Shoko.Server/API/v2/Modules/PlexWebhook.cs

```diff
@@ -247,26 +248,36 @@ public bool InvalidateToken()
     [HttpGet("sync")]
     public ActionResult Sync()
     {
-        Analytics.PostEvent("Plex", "SyncOne");
+        var user = HttpContext.GetUser();
+
+        _commandFactory.Create<CommandRequest_PlexSyncWatched>(c => c.User = user).Save();
 
-        _commandFactory.Create<CommandRequest_PlexSyncWatched>(c => c.User = HttpContext.GetUser()).Save();
         return APIStatus.OK();
     }
 
     [Authorize("admin")]
     [HttpGet("sync/all")]
     public ActionResult SyncAll()
     {
-        ShokoServer.Instance.SyncPlex();
+        var users = RepoFactory.JMMUser.GetAll()
+            .Where(user => !string.IsNullOrEmpty(user.PlexToken));
+
+        foreach (var user in users)
+            _commandFactory.Create<CommandRequest_PlexSyncWatched>(c => c.User = user).Save();
+
         return APIStatus.OK();
     }
 
     [Authorize("admin")]
-    [HttpGet("sync/{id}")]
-    public ActionResult SyncForUser(int uid)
+    [HttpGet("sync/{userID}")]
+    public ActionResult SyncForUser([FromQuery] int userID)
```

Comment:
You're changing the way this is working - is this a good idea at all? Users might be expecting a certain behaviour

Reply:
I can omit the changes I made to the v2 endpoints if you don't then to change.